config/src/token_remaps.rs

```diff
@@ -0,0 +1,158 @@
+    // from the default location
+    let map = if let Some(file) = remap_file {
+        parse_remap_from_file(file)
+    } else {
+        fetch_remap_from_repo(chain)
+    }?;
+
+    // Update the static token remap with the given one
+    let remap = map.to_remap();
+    TOKEN_REMAPS.set(remap).map_err(raw_err_str!("Failed to set token remap: {:?}"))
+}
+
+/// Parse a token remap from a JSON file
+fn parse_remap_from_file(file_path: String) -> Result<TokenRemap, String> {
+    // Read the file into a string
+    let file = std::fs::read_to_string(file_path)
+        .map_err(raw_err_str!("Failed to read remap file: {}"))?;
+    serde_json::from_str(&file).map_err(raw_err_str!("Failed to parse remap from file: {}"))
+}
+
+/// Pull the token remap from the repo
+fn fetch_remap_from_repo(chain: Chain) -> Result<TokenRemap, String> {
+    let url = format!("{}{}.json", REMAP_BASE_URL, chain);
+    let resp = reqwest::blocking::get(&url)
+        .map_err(raw_err_str!("Failed to fetch remap from repo: {}"))?;
+
+    resp.json().map_err(raw_err_str!("Failed to parse remap from Github: {}"))
+}
+
+#[cfg(test)]
+mod test {
+    use std::fs::File;
+
+    use arbitrum_client::constants::Chain;
+    use common::types::token::TOKEN_REMAPS;
+    use tempfile::{tempdir, TempDir};
+
+    use crate::token_remaps::parse_remap_from_file;
+
+    use super::{setup_token_remaps, TokenInfo, TokenRemap};
+
+    /// Get a temporary dir and remap file for testing
+    ///
+    /// Returns the temp dir, file, and file path
+    ///
+    /// The file and dir must be returned so they are not dropped
+    fn get_temp_dir() -> (TempDir, File, String) {
+        let dir = tempdir().unwrap();
+        let path = dir.path().join("remap.json").to_str().unwrap().to_string();
+        let file = File::create(&path).unwrap();
+
+        (dir, file, path)
+    }
+
+    /// Get a dummy remap and save it to a file
+    ///
+    /// Returns both the remap and the file path
+    ///
+    /// The dir must be created in the calling scope to ensure it is not dropped
+    fn gen_dummy_remap(file: &File) -> TokenRemap {
+        let remap = TokenRemap {
+            tokens: vec![TokenInfo {
+                name: "Renegade".to_string(),
+                ticker: "RNG".to_string(),
+                address: "0x1234".to_string(),
+                decimals: 18,
+            }],
+        };
+
+        // Write the remap to the file
+        serde_json::to_writer(file, &remap).unwrap();
+        remap
+    }
+
+    /// Tests parsing the token remap from a file
+    #[test]
+    fn test_parse_token_remap() {
+        let (_dir, file, path) = get_temp_dir();
+        let remap = gen_dummy_remap(&file);
+
+        // Parse the remap from the file
+        let parsed = parse_remap_from_file(path).unwrap();
+        assert_eq!(remap, parsed);
+    }
+
+    /// Tests that the token remap is correctly shared across threads
+    #[test]
```

Comment:
should we also set up a test for fetching the remap from githb?

Reply:
i don't want to require the unit tests to have an internet connection, so it's okay to ad hoc